src/interface/web/app/home/page.tsx

```diff
@@ -0,0 +1,463 @@
+            setProcessingMessage(false);
+        } else {
+            setMessage('');
+        }
+    }, [props.streamedMessages]);
+
+    const agents = data ? data.slice(0, 4) : []; //select first 4 agents to show as options
+    //generate colored icons for the selected agents
+    const icons = agents.map(agent => getIconFromIconName(agent.icon, agent.color) || <Image src={agent.avatar} alt={agent.name} width={50} height={50} />);
+    function fillArea(link: string, type: string, prompt: string) {
+        if (!link) {
+            let message_str = "";
+            prompt = prompt.charAt(0).toLowerCase() + prompt.slice(1);
+
+            if (type === "Online Search") {
+                message_str = "/online " + prompt;
+            } else if (type === "Paint") {
+                message_str = "/paint " + prompt;
+            } else {
+                message_str = prompt;
+            }
+
+            // Get the textarea element
+            const message_area = document.getElementById("message") as HTMLTextAreaElement;
+
+            if (message_area) {
+                // Update the value directly
+                message_area.value = message_str;
+                setMessage(message_str);
+            }
+        }
+    }
+    function handleAgentsClick(slug: string) {
+        return async () => {
+            setSelectedAgent(slug);
+            try {
+                const unauthenticatedRedirectUrl = `/login?next=/agents?agent=${slug}`;
+                const response = await fetch(`/api/chat/sessions?agent_slug=${slug}`, { method: "POST" });
+                if (!response.ok) {
+                    throw new Error(`HTTP error! status: ${response.status}`);
+                }
+                if (response.status == 200) {;
+                    highlightHandler(slug);
+                } else if (response.status == 403 || response.status == 401) {
+                    window.location.href = unauthenticatedRedirectUrl;
+                } else {
+                    alert("Failed to start chat session");
+                }
+            } catch (error) {
+                console.error("Error starting a conversation with the agent:", error);
+            }
+        };
+    }
+
+// Helper function to convert Tailwind color class to actual color value to prevent default border changes from being applied
+function tailwindColorToHex(colorClass: string): string {
+    const colorMap: { [key: string]: string } = {
+        'border-red-500': '#ef4444',
+        'border-blue-500': '#3b82f6',
+        'border-green-500': '#22c55e',
+        'border-yellow-500': '#eab308',
+        'border-purple-500': '#a855f7',
+        'border-pink-500': '#ec4899',
+        'border-indigo-500': '#6366f1',
+        'border-gray-500': '#6b7280',
+        'border-orange-500': '#f97316',
+    };
+    return colorMap[colorClass] || '#000000'; // Default to black if color not found
+}
+
+function highlightHandler(slug: string) {
```

Comment:
This can be simplified a lot by just using dynamic styling using this property. For the given agent card, check if the slug matched the selected agent. If so, restyle the border.